cdap-ui/app/features/apps/templates/list.html

```diff
@@ -4,9 +4,28 @@
     <h2> All Applications </h2>
   </div>
   <div class="col-xs-6 h2 text-right">
-    <my-file-select on-file-select="onFileSelected($files)"
-      data-button-label="Add App"
-      data-button-icon="fa-plus"> </my-file-select>
+    <div class="btn-group" dropdown="">
+      <button type="button" class="btn btn-default dropdown-toggle" dropdown-toggle="" aria-haspopup="true" aria-expanded="false">
+        Action <span class="caret"></span>
+      </button>
+      <ul class="dropdown-menu dropdown-menu-right" role="menu">
+        <li role="presentation">
+          <my-file-select on-file-select="onFileSelected($files)"
+            data-button-label="Add App"
+            data-button-icon="fa-plus"
+            data-dropdown="true"> </my-file-select>
+        </li>
+        <li role="presentation">
+          <a ng-click="caskConfirm()"
+             cask-confirmable="deleteApp(app.id)"
```

Comment:
In the context of "All Applications" app.id will be null. This is delete behavior/button should be added either to all the rows of the app or should atleast link to the admin page that allows the user to delete the app (which is the desired behavior I believe)